atomisticparsers/gromacs/parser.py

```diff
@@ -816,12 +816,13 @@ def parse_thermodynamic_data(self):
         n_frames = self.traj_parser.get('n_frames')
 
         # TODO read also from ene
+        # Make sure *.edr file is part of upload before attempting to parse it.
         edr_file = self.get_gromacs_file('edr')
-        self.energy_parser.mainfile = edr_file
-
-        # get it from edr file
-        if self.energy_parser.keys():
-            thermo_data = self.energy_parser
+        if edr_file:
+            self.energy_parser.mainfile = edr_file
+            # get it from edr file
+            if self.energy_parser.keys():
+                thermo_data = self.energy_parser
         else:
```

Comment:
What if you have an edr file but there are no keys? Then you won't end up in here! 

```python
if edr_file:
    if keys:

if not thermo_data:
    <try getting from input>
```

right?